stl/inc/atomic

```diff
@@ -639,18 +665,14 @@ struct _Atomic_storage<_Ty, 8> { // lock-free using 8-byte intrinsics
         // non-atomically initialize this atomic
     }
 
-#ifdef _M_IX86
-    void store(const _Ty _Value, const memory_order _Order = memory_order_seq_cst) noexcept {
-        // store with (effectively) sequential consistency
-        _Check_store_memory_order(_Order);
-        (void) exchange(_Value, _Order);
-    }
-#else // ^^^ _M_IX86 / !_M_IX86 vvv
-
     void store(const _Ty _Value) noexcept { // store with sequential consistency
         const auto _Mem           = _Atomic_address_as<long long>(_Storage);
         const long long _As_bytes = _Atomic_reinterpret_as<long long>(_Value);
-#ifdef _M_ARM64
+#if defined(_M_IX86)
+        _Compiler_barrier();
+        __iso_volatile_store64(_Mem, _As_bytes);
+        atomic_thread_fence(memory_order_seq_cst);

```

Comment:
As a non-`_Ugly` function call, our conventions technically require this to be qualified (even though there is no danger of ADL hijacking, since we control the argument):

```suggestion
        _STD atomic_thread_fence(memory_order_seq_cst);
```